src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -149,6 +149,11 @@ private void InitiateUnload()
                 _state = InternalState.Unloading;
             }
 
+            RemoveFromAllContexts();
+        }
+
+        internal void RemoveFromAllContexts()
+        {
             Dictionary<long, WeakReference<AssemblyLoadContext>> allContexts = AllContexts;
```

Comment:
This seems to be the only functional change here - that we now remove the ALC from this dictionary. But that dictionary only has a weak ref to the ALC - so how come it caused a memory leak?

Reply:
This is In Progress. There will be changes on the native side as well. The managed side by itself only leaks dictionary entries and weak references (and weak handles with them).